### ISSUE TITLE
Unable to start the server

### ISSUE THREAD

**Issue by mtoohey31** [gianarb/coc-grammarly#8]
Hello, I'm currently unable to successfully start the extension. Whenever I start editing a file that `coc-grammarly` is enabled for, I get an error stating:
```
[coc.nvim] The "coc-grammarly" server crashed 5 times in the last 3 minutes. The server will not be restarted.
```
`ps aux | grep grammarly` does find the process, but only for a split second as I open the file, then the process disappears again.
I noticed that this error shares some similarities with #7 so I wasn't sure whether to leave it as a reply or post it as a separate issue, so let me know if you'd like me to remove and reopen #7.
I've tried uninstalling and reinstalling multiple times and that doesn't resolve the issue. I've also tried building and adding the [latest grammarly release](https://github.com/znck/grammarly/releases/tag/v0.12.2) `server.js`, but that doesn't look like it was the issue because the output stays the same.
<details><summary>Here's the full logs</summary>
<p>

`ps aux` output:

```
mtoohey          30874  15.0  0.2  4549824  27808   ??  R    12:22pm   0:00.06 /usr/local/Cellar/node/15.4.0/bin/node /Users/mtoohey/.config/coc/extensions/node_modules/coc-grammarly/lib/server --node-ipc --clientProcessId=30840
```

`CocInfo` output:

```
## versions

vim version: NVIM v0.4.4
node version: v15.4.0
coc.nvim version: 0.0.79-7e550643a5
coc.nvim directory: /Users/mtoohey/.config/nvim/plugged/coc.nvim
term: iTerm.app
platform: darwin

## Log of coc.nvim

2020-12-17T12:36:06.593 INFO (pid:33619) [services] - registered service "eslint"
2020-12-17T12:36:06.637 INFO (pid:33619) [services] - registered service "coc-grammarly"
2020-12-17T12:36:06.638 INFO (pid:33619) [services] - coc-grammarly state change: stopped => starting
2020-12-17T12:36:06.838 INFO (pid:33619) [language-client-index] - coc-grammarly started with 33625
2020-12-17T12:36:07.017 INFO (pid:33619) [services] - coc-grammarly state change: starting => running
2020-12-17T12:36:07.020 INFO (pid:33619) [services] - service coc-grammarly started
2020-12-17T12:36:07.038 INFO (pid:33619) [services] - coc-grammarly state change: running => stopped
2020-12-17T12:36:07.038 INFO (pid:33619) [services] - coc-grammarly state change: stopped => starting
2020-12-17T12:36:07.043 INFO (pid:33619) [language-client-index] - coc-grammarly started with 33628
2020-12-17T12:36:07.053 INFO (pid:33619) [plugin] - coc.nvim 0.0.79-7e550643a5 initialized with node: v15.4.0 after 600ms
2020-12-17T12:36:07.057 INFO (pid:33619) [language-client-index] - cSpell started with 33629
2020-12-17T12:36:07.156 INFO (pid:33619) [services] - coc-grammarly state change: starting => running
2020-12-17T12:36:07.169 INFO (pid:33619) [services] - coc-grammarly state change: running => stopped
2020-12-17T12:36:07.169 INFO (pid:33619) [services] - coc-grammarly state change: stopped => starting
2020-12-17T12:36:07.172 INFO (pid:33619) [language-client-index] - coc-grammarly started with 33631
2020-12-17T12:36:07.260 INFO (pid:33619) [services] - coc-grammarly state change: starting => running
2020-12-17T12:36:07.272 INFO (pid:33619) [services] - coc-grammarly state change: running => stopped
2020-12-17T12:36:07.272 INFO (pid:33619) [services] - coc-grammarly state change: stopped => starting
2020-12-17T12:36:07.274 INFO (pid:33619) [language-client-index] - coc-grammarly started with 33632
2020-12-17T12:36:07.365 INFO (pid:33619) [services] - coc-grammarly state change: starting => running
2020-12-17T12:36:07.379 INFO (pid:33619) [services] - coc-grammarly state change: running => stopped
2020-12-17T12:36:07.379 INFO (pid:33619) [services] - coc-grammarly state change: stopped => starting
2020-12-17T12:36:07.389 INFO (pid:33619) [language-client-index] - coc-grammarly started with 33633
2020-12-17T12:36:07.511 INFO (pid:33619) [services] - coc-grammarly state change: starting => running
2020-12-17T12:36:07.525 INFO (pid:33619) [services] - coc-grammarly state change: running => stopped
```

</p>
</details>

Let me know if you'd like me to try any other debugging steps.

**Comment by marcoaaguiar** [comment]
I haven't checked, but I saw a similar issue with coc-pyright and node 15.4.0 (https://github.com/fannheyward/coc-pyright/issues/134#issuecomment-748418293)

**Comment by marcoaaguiar** [comment]
I have a guess on what might be the issue:

```bash
$ yarn
...
warning webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning webpack > watchpack > chokidar > fsevents@2.1.3: Please update to v 2.2.x
warning webpack > watchpack > watchpack-chokidar2 > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
...
```

You are running node v14+ on a Mac (fsevents is a dependency for Mac).
This might be breaking for you but not for me (Linux).
Could you investigate if, with earlier versions of npm, the plugin works?

@gianarb Are these dependencies necessary? Are they coming from upstream?
I haven't had the time to look into the code base but I would like to contribute in the future :)

Edit: coc-pyright have removed "vscode"-related dependencies recently and have a newer version of webpack. Would that help?

**Comment by mtoohey31** [comment]
Ok, I tried rolling back to node v12.20.0 and npm 4.6.1, and the coc-grammarly server doesn't display any errors, but it doesn't appear to do anything either, I wrote out the first section of the example in the readme and it didn't show anything, though CocInfo still seemed to indicate that it was running.
These older versions break many newer plugins, so I updated back to the current versions and coc-grammarly continued to not crash but do nothing.
I also tried it on a rasperry pi running manjaro that I had lying around, and on node v15.3.0 and it kept crashing on me there.
I've been playing with coc-spell-checker a little bit though and it seems to be serving my purposes, so don't feel the need to fix this issue on my account, but the fact that it has showed up for me on both computers at some point, it may likely show up for other people too.

**Comment by gianarb** [comment]
I can reproduce the issue, or at least this plugin does not work for me as well.

The version I currently pushed to master starts correctly but it does not do anything. Not sure why yet! I am not an expert and this project for me is an attempt to learn around this area. So not sure how much it will take for me to get something working again

**Comment by qhadron** [comment]
@gianarb It seems like the extension isn't always loaded, and it only loads for a few filetypes. If you set the filetype to `plaintext` (`setfiletype plaintext`), the plugin works. From what I can tell, the server only starts checking when the filetype is one of the ones in the `activationEvents` array: https://github.com/gianarb/coc-grammarly/blob/744ab1e720d8ce4bdf23cf9be35a6eb3882c5f76/package.json#L14-L17 .

**Comment by gianarb** [comment]
Ciao @qhadron right.
It starts for the one listed here https://github.com/gianarb/coc-grammarly/blob/master/src/index.ts#L20

**Comment by mtoohey31** [comment]
So I actually just managed to fix this error on my system and get it running, I've got no idea why what I did worked, but I'll put the details here in case anyone else is having this issue, the fix I used required that I downgrade node to 12.x though... so it's not ideal.

<details><summary>The Details...</summary>
<p>
I did some looking into how to debug coc-extensions correctly and realized I could use `:messages` in neovim to show some of the output of the extension, and it was outputting the error seen in this issue here: foreversd/forever/issues/1077, which is a 14.x error, so I downgraded to this nodejs version: https://archlinux.org/packages/community/x86_64/nodejs-lts-erbium which is 12.x.

Then I cloned this repository and built it with `npm install` and `npm run build`, then copied the new `lib` and `package.json` files into my `.config/coc/extensions/node_modules/coc-grammarly` directory. Then I downloaded [this specific release](https://github.com/znck/grammarly/releases/tag/v0.12.5) of grammarly, and ran `npm install` and `npm run build` for that too. Then I copied the `./out/server.js` file to `.config/coc/extensions/node_modules/coc-grammarly/dist/server.js`. And now it works!
</p>
</details>
